src/main/java/uk/gov/companieshouse/lookup/controller/CompanyLookupController.java

```diff
@@ -11,40 +11,66 @@
 import org.springframework.web.servlet.view.UrlBasedViewResolver;
 import org.springframework.web.util.UriTemplate;
-import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
 import uk.gov.companieshouse.lookup.exception.ServiceException;
+import uk.gov.companieshouse.lookup.model.CompanyConfirmation;
 import uk.gov.companieshouse.lookup.model.CompanyLookup;
 import uk.gov.companieshouse.lookup.service.CompanyLookupService;
+import uk.gov.companieshouse.lookup.validation.ValidationError;
+import uk.gov.companieshouse.lookup.validation.ValidationHandler;
 
 @Controller
-@RequestMapping("/accounts/lookup")
+@RequestMapping("/company-lookup/search")
 public class CompanyLookupController {
 
-    private static final UriTemplate FOUND_REDIRECT = new UriTemplate("/transaction/{transactionId}/confirmation");
+    private static final UriTemplate FOUND_REDIRECT = new UriTemplate(
+        "/company-lookup/{companyNumber}/confirmation");
+    private static final String COMPANY_LOOKUP = "lookup/companyLookup";
 
     @Autowired
     private CompanyLookupService companyLookupService;
 
-    @GetMapping
-    public String getCompanyLookup(Model model, HttpServletRequest request) throws ServiceException {
+    @Autowired
+    private ValidationHandler validationHandler;
 
-        model.addAttribute("companyLookup", new CompanyLookup());
-        model.addAttribute("backButton", "/nowhere");
-        return "lookup/companyLookup";
+    @GetMapping
+    public String getCompanyLookup(@RequestParam("forward") String forward, Model model) {
+        CompanyLookup companyLookup = new CompanyLookup();
+        model.addAttribute("companyLookup", companyLookup);
+        return COMPANY_LOOKUP;
     }
 
     @PostMapping
-    public String postCompanyLookup(@ModelAttribute("companyLookup") @Valid CompanyLookup companyLookup,
-        BindingResult bindingResult, Model model, HttpServletRequest request) throws ServiceException {
+    public String postCompanyLookup(@RequestParam("forward") String forward,
+        @ModelAttribute("companyLookup") @Valid CompanyLookup companyLookup,
+        BindingResult bindingResult, RedirectAttributes attributes) throws ServiceException {
 
-        CompanyProfileApi companyProfile = companyLookupService
-            .getCompanyProfile(companyLookup.getCompanyId());
+        List<ValidationError> validationErrors = companyLookupService
+            .validateCompanyLookup(companyLookup);
+        if (!validationErrors.isEmpty()) {
+            validationHandler.bindValidationErrors(bindingResult, validationErrors);
+            return COMPANY_LOOKUP;
+        }
 
-        model.addAttribute("companyProfile", companyProfile);
+        CompanyConfirmation companyConfirmation = companyLookupService
+            .getCompanyProfile(companyLookup.getCompanyNumber());
+
+        if (companyConfirmation == null) {
+            ValidationError error = new ValidationError();
```

Comment:
Can we have a method in the service / a validator class to construct this validation error?

Reply:
I thought this originally but it seemed a little excessive for four lines of code.